frontend/catalyst/jax_primitives.py

```diff
@@ -1518,14 +1518,14 @@ def _adjoint_lowering(
     # and classical arguments as-is, but substitute the quantum arguments with the arguments of the
     # block.
 
+    ctx = jax_ctx.module_context.context
     consts, cargs, qargs = tree_unflatten(args_tree, args)  # [1]
     _, _, aqargs = tree_unflatten(args_tree, jax_ctx.avals_in)  # [2]
 
     assert len(qargs) == 1, "We currently expect exactly one quantum register argument"
-    output_types = util.flatten(map(mlir.aval_to_ir_types, jax_ctx.avals_out))
 
     # Build an adjoint operation with a single-block region.
-    op = AdjointOp(output_types[0], qargs[0])
+    op = AdjointOp(ir.OpaqueType.get("quantum", "reg", ctx), qargs[0])
```

Comment:
Is there ever a case where `output_types[0]` not be `ir.OpaqueType.get("quantum", "reg", ctx)`?

Reply:
I think it is best to keep `output_types[0]` if possible as the value is derived from the program.

Reply:
Adjoint currently supports exactly one quantum register argument (and exactly one result). I used `output_types` before because I didn't see how to create types explicitly.

Reply:
I understand. I still believe that `output_types[0]` is better since it is the calculated output type as opposed to a declared output type. If `adjoint` were to other result types, we would need to change this bit of code back to `output_types[0]`.

Reply:
OK, no problem, will fix